server/dbSchemas/attendanceSchema.js

```diff
@@ -20,11 +32,49 @@ const courseSchema = new Schema({
         {
             date: Date,
             absent_students: [ { type: ObjectId, ref: 'user' } ],
-            seating_arrangement: [ { type: ObjectId, ref: 'user' } ]
+            seating_layout: { type: ObjectId, ref: 'seatingLayout', requried: true },
+            seating_arrangement: [[ { type: ObjectId, ref: 'user' } ]],
+            course_type: { type: String, required: true },
+            mandatory: { type: Boolean, requried: true }
         }
     ],
```

Comment:
So I don't think all of this data should be associated with day-to-day attendance. Remember that for the attendance we only need a list of the absent students and the date. Don't overthink it: we don't actually need to know where the students were that day, just whether they were absent or not. Therefore I think the `seating_layout`, `seating_arrangment`, `course_type`, and `mandatory` fields should be with the course itself rather than the day-to-day attendance. Those fields can change without having an attendance (imagine when the course is first created. Students will register and chose their seats and we will save that data not with a specific attendance day but with the course itself)

Reply:
I do agree that instead of saving the seating_layout in the attendance object they should be saved on the individual course.

Reply:
I agree that we could move `seating_layout` could be held in the course. However, moving `seating_arrangement` to the course would imply that students would sit in the same seat all semester. Which I don't think we want. But if I'm wrong, then we should definitely move it out of `attendance`.

I think we want to keep `course_type` inside the attendance so that teachers can specify which classes are a lecture, tutorial or lab. I also think we should keep `mandatory` in the the `attendance` so that teachers can specify which classes are mandatory or not, like how in SWE4103 some of our classes are mandatory and some aren't.

Reply:
For the `seating_arrangement` I see it as being for the course in the sense that we only need to keep the most up-to-date seating arrangement at any point in time. Meaning that every time a student changes their seat it will get changed in the `seating_arrangement` which is "global" for the course. We don't need to keep a record of where everyone was sitting for each day. When someone changes their seat we just change it and we are left with only one up-to-date seating arrangement. Does this make sense?

So the thing with the course type is that the tutorials/lectures/labs are going to be separate Courses in our system. For example, if Dawn wanted to represent SWE4103 in our app she would create 2 Courses (one for the lecture and one for the lab): "SWE4103_FR01A_LEC", "SWE4103_FR01A_LAB". Each of them will be their own "Course" entry in the database, which means the type is for the whole course rather than the day-to-day attendance. 

For the mandatory field, I'm still uncertain about how it is actually supposed to work, but I think it is meant to say whether all classes are mandatory or not. This interpretation means that it is a global setting for the course which is constant for the whole term rather than for each day. (but I could be wrong on this one, so I guess its ok to leave the `mandatory` here for now)

Reply:
I also see where I had a misunderstanding of "course_type", and I actually don't think it will need to be stored, the course name will be used to indicate the course type (so a SWE4103 lecture and SWE4103 lab should be stored as two separate courses in the system) according to the client's specifications. 
I believe the "mandatory" field is where it should be.
For the "seating_arrangement", if we were to move it to the course, could we not update it using backend when students update/change seats? And then that would make "attendance" simpler, because we would only need to store a 1D array of User objects (for the absent students)?